modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/applications/forms/steps/ContactInfo/ContactInfoForm.tsx

```diff
@@ -23,72 +23,62 @@ import ClayIcon from '@clayui/icon';
 import classNames from 'classnames';
 import {useContext, useEffect, useState} from 'react';
 
+import {useLocation} from '../../../../../common/hooks/useLocation';
+import {CONSTANTS} from '../../../../../common/utils/constants';
 import {
 	ACTIONS,
 	NewApplicationAutoContext,
 } from '../../../context/NewApplicationAutoContextProvider';
+import {HasValidationsTypes, hasRequiredErrorTypes} from '../../Types';
+import {
+	isValidEmail,
+	isValidPhone,
+	isValidUSZip,
+} from '../../utils/formValidations';
 
 const ContactInfo = () => {
 	const [state, dispatch] = useContext(NewApplicationAutoContext);
 
+	const {setAutoComplete} = useLocation();
+
+	const addressElement = document.querySelector(
+		'#streetAddress'
+	) as HTMLInputElement;
+
 	const {form} = state?.steps?.contactInfo;
 
+	const updateFormWithGoogleAddress = (address: any) => {
+		const payload = {
+			...form,
+			city: address.city,
+			state: address.state,
+			streetAddress: address.street,
+			zipCode: address.zip,
+		};
+
+		dispatch({payload, type: ACTIONS.SET_CONTACT_INFO_FORM});
+	};
+
 	const handleChangeField = (fieldName: string, fieldValue: string) => {
 		const payload = {
 			...form,
 			[fieldName]: fieldValue,
 		};
 
+		if (addressElement) {
+			setAutoComplete(addressElement, updateFormWithGoogleAddress);
+		}
+
 		dispatch({payload, type: ACTIONS.SET_CONTACT_INFO_FORM});
 		dispatch({payload: true, type: ACTIONS.SET_HAS_FORM_CHANGE});
 	};
 
-	const stateOptions = [
-		{
-			label: '',
-			value: '',
-		},
-		{
-			label: 'CHOOSE AN OPTION',
-			value: 'CHOOSE AN OPTION',
-		},
-		{
-			label: 'CA',
-			value: 'CA',
-		},
-		{
-			label: 'NV',
-			value: 'NV',
-		},
-		{
-			label: 'NY',
-			value: 'NY',
-		},
-	];
-
-	type hasRequiredErrorTypes = {
-		city: boolean;
-		dateOfBirth: boolean;
-		email: boolean;
-		firstName: boolean;
-		lastName: boolean;
-		phone: boolean;
-		state: boolean;
-		streetAddress: boolean;
-		zipCode: boolean;
-	};
-
-	type HasValidationsTypes = {
-		city: boolean;
-		dateOfBirth: boolean;
-		email: boolean;
-		firstName: boolean;
-		lastName: boolean;
-		phone: boolean;
-		state: boolean;
-		streetAddress: boolean;
-		zipCode: boolean;
-	};
+	useEffect(() => {
```

Comment:
I believe you don't need this ```useEffect``` anymore

Reply:
I believe that we still need it. We can discuss it better :)